cerealize/exceptions.h

```diff
@@ -33,6 +33,8 @@ struct ParseJSONException : Exception {
   explicit ParseJSONException(const std::string& input) : Exception("Invalid JSON:\n" + input) {}
 };
 
+struct CerealizeFileStreamErrorException : Exception {};
```

Comment:
This might be a longer conversation, but I'd probably make this exception derive from `FileException`.

Effectively, the only way it might occur is a corrupted file. What do you think?